### PR TITLE
Update validate plugin entry to account for publishing on behalf of orgs

### PR DESCRIPTION
**Pull request by dvargas92495**
This change should allow plugin authors to publish plugins on behalf of their GitHub organization

### CODE REVIEW

File: .github/workflows/validate-plugin-entry.yml [hunk 1]
```diff
@@ -50,7 +50,16 @@ jobs:
             console.log(`Repo info: ${owner}/${repo}`);
 
             if (owner.toLowerCase() !== author.toLowerCase()) {
-              addError(`The newly added entry is not at the end, or you are submitting on someone else's behalf. Last plugin in the list is: ${plugin.repo}`);
+              const onBehalfOfOrg = await octokit.rest.orgs
```
Comment (joethei) [inline .github/workflows/validate-plugin-entry.yml:53]: This will break, on GH actions it would be `github.rest...`, will work the same though.

**Review by joethei — CHANGES_REQUESTED** [review]

File: .github/workflows/validate-plugin-entry.yml (hunk 1 above)
Comment (dvargas92495) [reply .github/workflows/validate-plugin-entry.yml:53]: gah sorry forgot to swap it out from my internal testing

**Comment by lishid** [comment]
Hmm according to the github docs, it seems to care whether the requester is the org member:
https://docs.github.com/en/rest/orgs/members?apiVersion=2022-11-28#check-organization-membership-for-a-user

Seems like it returns 302 if the requester isn't and 204 if both the requester and the username passed in is a member. Does anyone know how octokit handles the differentiation between 302 and 204?

**Comment by dvargas92495** [comment]
302s seem to just return the data, which aligns with the current `then => true, catch => false` logic

https://github.com/octokit/request.js/blob/master/src/fetch-wrapper.ts#L69-L118

**Comment by dvargas92495** [comment]
I just did

```
.checkMembershipForUser({
  org: "obsidian-community",
  username: "joethei",
})
```

and it actually returned a 204 locally, despite me not being in the obsidian-community org. In any case, I believe the Promise should resolve

**Comment by lishid** [comment]
Ok! Let's merge and see. Thank you for the contribution ❤️

**Comment by dvargas92495** [comment]
Seems to have worked!

https://github.com/obsidianmd/obsidian-releases/pull/1342

**Pull request merged**